src/analyzer/protocol/tcp/TCP.cc

```diff
@@ -2174,3 +2179,192 @@ void TCPStats_Analyzer::DeliverPacket(int len, const u_char* data, bool is_orig,
 	else
 		resp_stats->DataSent(network_time, seq, len, caplen, data, ip, 0);
 	}
+
+TCPRS_Analyzer::TCPRS_Analyzer(Connection* conn)
+: TCP_ApplicationAnalyzer("TCPRS", conn)
+{
+}
+
+TCPRS_Analyzer::~TCPRS_Analyzer()
+{
+    delete orig;
+    delete resp;
+}
+
+void TCPRS_Analyzer::Init()
+{
+  TCP_ApplicationAnalyzer::Init();
+
+  measurement_location = NO_LOCATION_DATA;
+  max_confidence = 0;
+  location_confidence = 0.0;
+  sample_count = 0;
+
+  orig = new TCPRS_Endpoint(TCP()->Orig(), this);
+  resp = new TCPRS_Endpoint(TCP()->Resp(), this);
```

Comment:
This might be one of few points where they interact. I tried to make this as independent from the TCP analyzer as possible. There might actually be little work in turning this into a separate plugin.